### PR TITLE
Docs: Fix conflicting description on primary key support in Flink

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

**Comment by manuzhang** [comment]
@hililiwei please help review.

**Comment by hililiwei** [comment]
Thanks for the contribution. Can you add primary key instructions and examples above?

**Comment by manuzhang** [comment]
@hililiwei I added examples, merged `PARTITIONED BY` section into `CREATE TABLE` and fixed indent in several places.

**Comment by manuzhang** [comment]
@hililiwei gentle ping for another review.

File: docs/flink-ddl.md [hunk 1]
```diff
@@ -159,29 +162,16 @@ Table create commands support the commonly used [Flink create clauses](https://n
 * `COMMENT 'table document'` to set a table description.
 * `WITH ('key'='value', ...)` to set [table configuration](../configuration) which will be stored in Iceberg table properties.
 
-Currently, it does not support computed column, primary key and watermark definition etc.
-
-### `PARTITIONED BY`
```
Comment (hililiwei) [inline docs/flink-ddl.md:164]: I don’t recommend deleting this part, we can configure it as a child of the cteate table like spark DDL does.

File: docs/flink-ddl.md [hunk 2]
```diff
@@ -159,29 +162,16 @@ Table create commands support the commonly used [Flink create clauses](https://n
 * `COMMENT 'table document'` to set a table description.
```
Comment (hililiwei) [inline docs/flink-ddl.md:162]: I recommend describe "PRIMARY KEY" here, and add a separate section to explain it, like https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/dev/table/sql/create/#primary-key

File: docs/flink-ddl.md [hunk 3]
```diff
@@ -144,44 +144,61 @@ CREATE DATABASE iceberg_db;
...
-    data STRING
-);
+    `data` STRING NOT NULL
+) 
+WITH ('format-version'='2');
 ```
 
 Table create commands support the commonly used [Flink create clauses](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/create/) including:
 
-* `PARTITION BY (column1, column2, ...)` to configure partitioning, Flink does not yet support hidden partitioning.
```
Comment (hililiwei) [inline docs/flink-ddl.md:158]: why delete this?

File: docs/flink-ddl.md [hunk 4]
```diff
@@ -144,44 +144,61 @@ CREATE DATABASE iceberg_db;
...
 To create a partition table, use `PARTITIONED BY`:
 
 ```sql
 CREATE TABLE `hive_catalog`.`default`.`sample` (
-                                                   id BIGINT COMMENT 'unique id',
-                                                   data STRING
-) PARTITIONED BY (data);
+    id BIGINT COMMENT 'unique id',
+    `data` STRING NOT NULL,
+    PRIMARY KEY(`id`) NOT ENFORCED
```
Comment (hililiwei) [inline docs/flink-ddl.md:186]: the partition fields should be included in primary key, or we can delete this 186 line.

File: docs/flink-ddl.md (hunk 3 above)
Comment (manuzhang) [reply docs/flink-ddl.md:158]: there is no partition by in the above example

**Comment by manuzhang** [comment]
@hililiwei any more comments?

**Comment by manuzhang** [comment]
@hililiwei gentle ping for review.

**Comment by manuzhang** [comment]
@hililiwei @nastra I've rebased on latest master changes. Please help review again.

File: docs/flink-ddl.md [hunk 5]
```diff
@@ -169,13 +184,15 @@ To create a partition table, use `PARTITIONED BY`:
...
+    `data` STRING NOT NULL
+) 
+PARTITIONED BY (data) 
+WITH ('format-version'='2');
 ```
 
-Iceberg support hidden partition but Flink don't support partitioning by a function on columns, so there is no way to support hidden partition in Flink DDL.
+Iceberg supports hidden partition but Flink doesn't support partitioning by a function on columns, so there is no way to support hidden partition in Flink DDL.
 
-### `CREATE TABLE LIKE`
```
Comment (bitsondatadev) [inline docs/flink-ddl.md:178]: Same comment here for below. Let's keep this focused on fixing these descriptions and do this in a larger PR.

**Review by bitsondatadev — CHANGES_REQUESTED** [review]
Overall looks good, I would just remove the irrelevant heading changes and we'll save that for another PR.

File: docs/flink-ddl.md [hunk 6]
```diff
@@ -145,13 +145,14 @@ CREATE DATABASE iceberg_db;
 USE iceberg_db;
 ```
 
-### `CREATE TABLE`
```
[the remark this answers is not in the document]
Comment (manuzhang) [reply docs/flink-ddl.md:148]: The reason to fix in this PR is I need to create two sub-sections under `CREATE TABLE`.

[the remark this answers is not in the document]
Comment (manuzhang) [reply docs/flink-ddl.md:148]: @bitsondatadev what do you think?

[the remark this answers is not in the document]
Comment (bitsondatadev) [reply docs/flink-ddl.md:148]: I agree with the change I just do t think it should be in this PR since it's not related.

**Comment by manuzhang** [comment]
@bitsondatadev Comments addressed. Please take another look.

File: docs/flink-ddl.md [hunk 7]
```diff
@@ -138,6 +138,7 @@ Using `-i <init.sql>` option to initialize SQL Client session:
 
 ### `CREATE DATABASE`
 
+By default, Iceberg will use the `default` database in Flink. Using the following example to create a separate database in order to avoid creating tables under the `default` database:
```
Comment (bitsondatadev) [inline docs/flink-ddl.md:141]: This looks like duplicated text. Could you please remove?

**Review by bitsondatadev — APPROVED** [review]
Looks good % my comments about the duplicate text. Please update that.

My pass was just basic. I'm not a Flink expert so I'll push one someone else to review this once the last change is addressed.

**Comment by bitsondatadev** [comment]
@stevenzwu, please review this at your earliest convenience.

**Comment by manuzhang** [comment]
@hililiwei @nastra @stevenzwu @bitsondatadev ping again.

**Review by hililiwei — APPROVED** [review]

File: docs/flink-ddl.md [hunk 8]
```diff
@@ -160,20 +161,36 @@ Table create commands support the commonly used [Flink create clauses](https://n
...
+
+#### `PRIMARY KEY`
+
+Primary key constraint can be declared for a column or a set of columns, which must be unique and do not contain null.
+It's required for [`UPSERT` mode](../flink/flink-writes.md#upsert).
+
+```sql
+CREATE TABLE `hive_catalog`.`default`.`sample` (
+    id BIGINT COMMENT 'unique id',
+    `data` STRING NOT NULL,
```
Comment (nastra) [inline docs/flink-ddl.md:174]: what's the reason for updating data to \`data\`? If it's required, then we should maybe update the entire file as there's one more example that hasn't been updated

File: docs/flink-ddl.md [hunk 9]
```diff
@@ -160,20 +161,36 @@ Table create commands support the commonly used [Flink create clauses](https://n
...
 To create a partition table, use `PARTITIONED BY`:
 
 ```sql
 CREATE TABLE `hive_catalog`.`default`.`sample` (
     id BIGINT COMMENT 'unique id',
-    data STRING
-) PARTITIONED BY (data);
+    `data` STRING NOT NULL
+) 
+PARTITIONED BY (data)
```
Comment (nastra) [inline docs/flink-ddl.md:189]: should "data" be highlighted the same way in the `PARTITIONED BY` section?

Comment (manuzhang) [reply docs/flink-ddl.md:189]: I've removed unnecessary highlights which were copied from `flink-writes.md`.

File: docs/flink-ddl.md [hunk 10]
```diff
@@ -160,20 +161,36 @@ Table create commands support the commonly used [Flink create clauses](https://n
...
 To create a partition table, use `PARTITIONED BY`:
 
 ```sql
 CREATE TABLE `hive_catalog`.`default`.`sample` (
     id BIGINT COMMENT 'unique id',
-    data STRING
-) PARTITIONED BY (data);
+    data STRING NOT NULL
+) 
+PARTITIONED BY (data)
```
Comment (nastra) [inline docs/flink-ddl.md:189]: I think the `PARTITIONED BY (data) WITH ('format-version'='2');` should be appended to the previous line. Same for the other places that were updated

Comment (manuzhang) [reply docs/flink-ddl.md:187]: I've put `with` on the same line as `)` in other places but for `PARTITIONED BY` it's on a separate line if we follow [Flink's doc](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/create/)

**Review by nastra — APPROVED** [review]
LGTM, thanks @manuzhang

**Pull request merged**